controllers/azurejson_machine_controller.go

```diff
@@ -216,7 +217,7 @@ func (r *AzureJSONMachineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		// TODO: remove this ClientID lookup code when the fixed cloud-provider-azure is default
 		idsClient := identities.NewClient(clusterScope)
 		userAssignedIdentityIfExists, err = idsClient.GetClientID(
-			ctx, azureMachine.Spec.UserAssignedIdentities[0].ProviderID)
+			ctx, strings.TrimPrefix(azureMachine.Spec.UserAssignedIdentities[0].ProviderID, "azure://"))
```

Comment:
can the azure prefix be a const?

Reply:
It already is: `ProviderIDPrefix` in azure/defaults. Will update.